pkg/loop/internal/pb/relayer.proto

```diff
@@ -0,0 +1,306 @@
+message PluginArgs {
+  string transmitterID = 1;
+  bytes pluginConfig = 2;
+}
+
+// NewConfigProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewConfigProvider].
+message NewConfigProviderRequest {
+  RelayArgs relayArgs = 1;
+}
+
+// NewConfigProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewConfigProvider].
+message NewConfigProviderReply {
+  uint32 configProviderID = 1;
+}
+
+// NewMedianProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMedianProvider].
+message NewMedianProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+// NewMedianProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMedianProvider].
+message NewMedianProviderReply {
+  uint32 medianProviderID = 1;
+}
+
+// NewMercuryProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMercuryProvider].
+message NewMercuryProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+// NewMercuryProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMercuryProvider].
+message NewMercuryProviderReply {
+  uint32 mercuryProviderID = 1;
+}
+
+// ChainStatusRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatus].
+message ChainStatusRequest {
+  string id = 1;
+}
+
+// ChainStatusReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatus].
+message ChainStatusReply {
+  ChainStatus chain = 1;
+}
+
+// ChainStatusesRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatuses].
+message ChainStatusesRequest {
+  int32 offset = 1;
+  int32 limit = 2;
+}
+
+// ChainStatusesReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatuses].
+message ChainStatusesReply {
+  repeated ChainStatus chains = 1;
+  int32 count = 2; // total count of chains
+}
+
+// ChainStatus represents [github.com/smartcontractkit/chainlink-relay/pkg/types.ChainStatus].
+message ChainStatus {
+  string id = 1;
+  bool enabled = 2;
+  string config = 3; // TOML
+}
+
+// NodeStatusesRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NodeStatuses].
+message NodeStatusesRequest {
+  int32 offset = 1;
+  int32 limit = 2;
+  repeated string chainIDs = 3;
+}
+
+// NodeStatusesReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NodeStatuses].
+message NodeStatusesReply {
+  repeated NodeStatus nodes = 1;
+  int32 count = 2; // total count of nodes
+}
+
+// NodeStatus represents [github.com/smartcontractkit/chainlink-relay/pkg/types.NodeStatus].
+message NodeStatus {
+  string chainID = 1;
```

Comment:
Should we think about making chainID a separate proto?

It could then have structure like:

message ChainID {
    string chainFamily = 1;
    string id = 2;   // Unique id within the chain
    (optional) string globalId = 3;   // A globally unique string id for the chain
}

Reply:
We don't currently ever mix nodes from different families together, so this distinction is not necessary. I can see how it could be useful in the future, but I would prefer different terminology for the combination of fields than reusing `ChainID`.

Reply:
So this will be a string representation of plugin's chain family id?
For example for EVM, it will be int to string?

Maybe I'll need to look into code more deeply, but how would the Core Node understand the native chain family's string chainId? I assume it needs to keep an index/mapping somewhere.

Reply:
Everything is reduced to `string` _because_ the core node can't/shouldn't understand it. We already have to use string representations for TOML, go maps, etc. so this is was just a simplification.